vllm/forward_context.py

```diff
@@ -78,10 +78,10 @@ def num_tokens_across_dp(num_tokens: int, dp_size: int,
         num_tokens_across_dp = [0] * dp_size
         num_tokens_across_dp[dp_rank] = num_tokens
         num_tokens_tensor = torch.tensor(num_tokens_across_dp,
-                                         device="cpu",
+                                         device="cuda",
```

Comment:
I think it needs to be:
```suggestion
                                         device=current_platform.device_type,
```